### ISSUE TITLE
[Marketing] Branded Admin Exprerience

### ISSUE THREAD

**Issue by jonrz** [joomla/joomla-ux#24]
To improve both corporate market penetration, and sense of ownership, Joomla need to offer a branded experience.

Both the client and the integrator will have visible space in the login and admin interface, with a visual balance of: Joomla 50%, Client 40%, and Integrator 10%.

**Comment by brianteeman** [comment]
this already exists and has done for twelve years

**Comment by jonrz** [comment]
As an incomplete tool, yes.
We want to give it a proper space, and rules, as well as proper visual balance.
Integrators won't be allowed to add logo, just text and a link (can be to their site, or support page).

Rules would be:
1. If a human changed the data, it can only be changed again by a human.
2. Templates (on install or update) can change integrator data if rule 1 allows.
3. No extension is allowed to change client or integrator data in any case.

For that we need a separate flag for "manually changed" in each config (client and integrator).

I'm also considering requiring/mandating a transparent background monochrome PNG for the client logo, to maintain proper visual balance and avoid ugly admin areas, but I'm not sure that can be technically enforceable (can we detect the composition of the PNG)?

This is backed both by the UX team and the Marketing team, and will be implemented, we just need to find the better way to do this.

**Comment by brianteeman** [comment]
As GPL software you have the freedom to change the code. This would also mean you can not put restrictions in place such as those you mention above. Any attempt to put restrictions with what you can do with joomla go against the license and the spirit of joomla.

**Comment by mbabker** [comment]
The template can be opinionated on how it will support features (i.e. the monochrome thing), but in general the full API could not be feasibly written in a way to support those rules as defined.

Also remember that extensions have full access to the filesystem and database (even if they bypass the Joomla API and go straight for the PHP native stuff).  So anyone can do anything they want outside of the Joomla API and impact things inside the application.  That part makes enforcement of the rules semi impractical.

**Comment by brianteeman** [comment]
there are so many things that other teams are waiting for answers for from the UX team can you please prioritise those before starting anything new. So many people's work is on hold until the UX team makes a decision on them its really a bottleneck

**Comment by Sandra97** [comment]
Yes, we can currently replace the backend logo and the login logo by anything you want. 
Jonathan's idea is to keep Joomla! logo + add the option to display in a smaller size the logo of the client as well as name/link of the integrator. Which is for me a good idea.

**Comment by brianpeat** [comment]
I personally think the only way we should be putting restrictions on extensions and templates is if they want to be in our directories. If someone doesn't care about our standards enough to be in the JED or new Template directory, then they likely won't care what rules we set up anyway. At least with the directories we can create a set of standards they must meet if they want to be in the directory. It also gives users a sense of security knowing the extensions play by the rules.

We could even set rules on the resource directory. You want to be listed, you sign/agree to follow some rules about site building. That one probably would get rejected, but it's the only way I know to force site builders to adhere to some sort of rules.

As far as changing the logo, I'm pretty sure the control panel joomla logo always stays put. It's the Joomla logo on the right side that gets replaced. So in a sense some tiny bit of branding always stays unless you override it.

**Comment by rgmears** [comment]
Two things are being overlooked here.
1) Anyone can replace the Joomla logo image with their own by using the image name and uploading a different image.
2) The grey-scale idea (from Jonathan) can be controlled with CSS. Although, even that can be changed.

I agree with Brian T. This is a low priority item.

**Comment by jonrz** [comment]
@brianpeat idea for restriction based on listing in the extensions directory covers everything.
@mbabker Can we do that?

@rgmears it's not grayscale, its monochrome over transparent bg, which is different. We can't do that with CSS and expect a good result.
As for anyone can replace the logo, nothing we can do about it.
We are providing a tool for them to change it easily and withing a reasonable set of guidelines.

@brianteeman I know, and we are working on answers for a lot of things. This issue came up, and we are talking about it now, so we can implement it later.

**Comment by mbabker** [comment]
We cannot do something that mandates the ability to check if an extension is JED listed.  This would limit feature usage based on running your install on an intranet or with non-distributed custom extensions.

**Comment by jonrz** [comment]
OK, I still think it is good to implement this system, even if restrictions are non enforceable.

@Sandra97 are you OK with this, knowing this can be hijacked by any template or pervasive extension?
Can we at least establish documented guidelines for template/extension providers (hoping they will follow, and knowing we can't do anything if they don't)?

**Comment by Sandra97** [comment]
Anyone is free to change the code. So no reason (and no way) to have "rules". 
You can of course establish a document with guidelines, even if in the end everyone can do whatever they want.  ;)

**Comment by brianpeat** [comment]
My suggestion wasn't to check if an extension is JED listed, it was to set rules that have to be agreed to and followed if your extension is to be listed in the JED. Then we post the standards so people know "these are the rules that are agreed to to be listed." If users don't care, that's fine, but if they do care, it at least shows we have some standards for the being listed in the JED. It's really the only way I see of "enforcing" some standards and not breaking the spirit of open source. Personally I don't see the need for any of this, but if you DO want to enforce something, this seems like the logical way to do it. If someone doesn't want to be in the JED, then they can do whatever the heck they want when their extension or template gets installed.
